swig/cpp/src/Struct.h

```diff
@@ -70,19 +70,19 @@ class Val
     Val(sr_val_t *val, S_Counter counter);
     Val(const char *val, sr_type_t type = SR_STRING_T);
     Val(bool bool_val, sr_type_t type = SR_BOOL_T);
-    Val(double decimal64_val, sr_type_t type = SR_DECIMAL64_T);
```

Comment:
The problem was that python and lua do not recognize between int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t, uint64_t and double, it is the same type for them.

swig simply overloads the constructors and functions. So now there are two constructors/functions for numbers , one for double and one for integers (int8_t, ..). So their type signature need to be different that is why in swig there is now:

Val(double decimal64_val);
Val(uint64_t uint64_val, sr_type_t type)

As you can see swig puts a few limitations on your code.

If you have any better solution I am open to change.